cmd/utils/flags.go

```diff
@@ -1169,10 +1169,9 @@ func SetP2PConfig(ctx *cli.Context, cfg *p2p.Config) {
 	lightClient := ctx.GlobalString(SyncModeFlag.Name) == "light"
 	lightServer := ctx.GlobalInt(LightServeFlag.Name) != 0
 
-	lightPeers := 0
-	if ctx.GlobalIsSet(LightMaxPeersFlag.Name) {
-		lightPeers = ctx.GlobalInt(LightMaxPeersFlag.Name)
-	}
+	// LightPeers is in the eth config, not the p2p config, so we don't have it here, so we
+	// get it here separately using GlobalInt()
+	lightPeers := ctx.GlobalInt(LightMaxPeersFlag.Name)
```

Comment:
shouldn't we use `IsSet` to verify is the flag if configured? If it's not configured, then use what's on the cfg already that comes from toml config and the default config

Reply:
Using the value from `cfg` would be the most correct thing to do.  The complication is that they're different configs.  There is `eth.Config`, which is where `LightPeers` is (as well as `LightServ`), and there is `p2p.Config` which is what is used in this function.  And, in fact, this function is called before the `eth.Config` struct is generated (by `utils.SetEthConfig()`), so we'd have to confirm it's ok to reorder them.  The top-level structure can seen here: https://github.com/celo-org/celo-blockchain/blob/master/cmd/geth/config.go#L106-L128

I do think it's problematic how this is done, and in fact there's at least one other difficulty I know of due to the fact that `LightPeers` and `MaxPeers` are not in the same config structure, which is that `p2p/server.go` for its `MaxDialedConns()` calculation doesn't take into account how many of the peers under `MaxPeers` are really supposed to be light clients (and so would be inbound connections, not dialed connections).  So I do think this is something we should change, but not as part of this PR, since it involves a larger scope.

In the absence of access to the `eth.Config`, I think this is the best we can do, and the only real drawback is it would miss the LightPeers value coming from TOML if there is one.  On a similar note, the `lightServer` var (https://github.com/celo-org/celo-blockchain/blob/master/cmd/utils/flags.go#L1170) above also relies on just the CLI flags and would miss it if we specified `LightServ` via TOML and think we're not running light server even though we are (because that is ultimately determined by `eth.Config` elsewhere, which does respect the TOML).  Both of these are also bugs upstream.  It seems that go-ethereum's support for TOML is not perfect either.

Reply:
Ok, so maybe something to work while improving/fxing our TOML support?

Can we add it to that issue then and resolve here?